client/beefy/src/worker.rs

```diff
@@ -910,6 +901,73 @@ where
 			}
 		}
 	}
+
+	/// Report the given equivocation to the BEEFY runtime module. This method
+	/// generates a session membership proof of the offender and then submits an
+	/// extrinsic to report the equivocation. In particular, the session membership
+	/// proof must be generated at the block at which the given set was active which
+	/// isn't necessarily the best block if there are pending authority set changes.
+	pub(crate) fn report_equivocation(
+		&self,
+		proof: EquivocationProof<NumberFor<B>, AuthorityId, Signature>,
+	) -> Result<(), Error> {
+		let rounds =
+			self.persisted_state.voting_oracle.active_rounds().ok_or(Error::UninitSession)?;
+		let (validators, validator_set_id) = (rounds.validators(), rounds.validator_set_id());
+
+		if proof.set_id != validator_set_id {
+			debug!(target: "beefy", "🥩 Skip equivocation report for old set id: {:?}", proof.set_id);
+			return Ok(())
+		} else if let Some(local_id) = self.key_store.authority_id(validators) {
+			if proof.equivocation.id == local_id {
+				debug!(target: "beefy", "🥩 Skip equivocation report for own equivocation");
+				return Ok(())
+			}
+		}
+
+		let number = proof.equivocation.round_number;
+		let hash = self
```

Comment:
Dumb q: why do you need this hash? Can we just pass `BlockId::Number(number)` to the `generate_key_ownership_proof` call?